src/reducers/data_reducer.js

```diff
@@ -1,13 +1,22 @@
-import { FETCH_DATA } from './../actions/types';
+import { FETCHED_DATA, START_FETCHING } from './../actions/types';
 
-const InitialState = [];
+const InitialState = {
+  isFetching: false,
+  categories: [],
+};
 
 export default (state = InitialState, action) => {
   switch (action.type) {
-    case FETCH_DATA:
-      {
-        return action.payload;
-      }
+    case FETCHED_DATA:
+      return Object.assign({}, state, {
+        categories: action.payload,
+        isFetching: false,
+      });
+    case START_FETCHING:
+      return Object.assign({}, state, {
+        categories: [],
```

Comment:
1. No need to set categories to []
2. See if you can get object spread operator to work. It would be cleaner if we could just user 
```
return { ...state, categories: [], isFeching: true } 
```
for example, instead of Object.assign which is old school.